src/hooks/pg.ts

```diff
@@ -0,0 +1,32 @@
+import type pg from "pg";
+
+import { recording } from "../recorder";
+
+type HasQueryType = typeof pg.Client | typeof pg.Pool;
+type HasQuery = InstanceType<HasQueryType>;
+
+function createProxy<T extends HasQueryType>(hasQueryType: T) {
+  return new Proxy(hasQueryType, {
+    construct(target, argArray, newTarget) {
+      const hasQuery = Reflect.construct(target, argArray, newTarget) as HasQuery;
+      const f = hasQuery.query.bind(hasQuery);
+      hasQuery.query = new Proxy(f, {
+        apply(target, thisArg, argArray: Parameters<typeof f>) {
+          recording.sqlQuery("postgresql", argArray[0]);
+          return target.apply(thisArg, argArray);
+        },
+      });
+      return hasQuery;
+    },
+  });
+}
+
+export default function pgHook(mod: typeof pg) {
+  mod.Client = createProxy(mod.Client);
+  mod.Pool = createProxy(mod.Pool);
```

Comment:
Is there a reason to hook the construction and create a new proxy for every object instead of just hooking `mod.{Client,Pool}.prototype.query`?